### PR TITLE
Remove Deprecations.swift and fix warnings

### PR DESCRIPTION
**Pull request by Jon889**
The deprecations file causes a lot of warnings in Xcode 10.2. I've bumped the major version to 6.0.0 as this breaks the public API. There's been plenty of time for people to stop using the deprecated methods.

### CODE REVIEW

File: Sources/JSONUtilities/JSONFileLoading.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ public extension Dictionary where Key: StringProtocol, Value: Any {
    
    - returns: An initilized JSONDictionary
    */
-  public static func from(filename: String, bundle: Bundle = .main) throws -> JSONDictionary {
+  static func from(filename: String, bundle: Bundle = .main) throws -> JSONDictionary {
```
Comment (lucianomarisi) [inline Sources/JSONUtilities/JSONFileLoading.swift:39]: [change request] Could we keep the `public` access control here?

File: Sources/JSONUtilities/JSONFileLoading.swift [hunk 2]
```diff
@@ -71,7 +71,7 @@ public extension Dictionary where Key: StringProtocol, Value: Any {
    
    - returns: An initilized JSONDictionary
    */
-  public static func from(jsonData: Data) throws -> JSONDictionary {
+  static func from(jsonData: Data) throws -> JSONDictionary {
```
Comment (lucianomarisi) [inline Sources/JSONUtilities/JSONFileLoading.swift:74]: [change request] Could we keep the `public` access control here?

**Review by lucianomarisi — COMMENTED** [review]
Thanks for the changes, just added a minor change request before we merge it.

**Comment by Jon889** [comment]
I removed the public access control keywords because Xcode 10.2 added a warning for when its redundant. (Methods in extensions follow the access of the extension by default, rather than internal by default). Personally I prefer being explicit with it, but I think it’s better to have zero warnings.

**Comment by Jon889** [comment]
Thanks for merging it :) How are the releases handled?

**Comment by lucianomarisi** [comment]
All tagged and released on Cocoapods.

**Comment by Jon889** [comment]
Thanks!

**Pull request merged**